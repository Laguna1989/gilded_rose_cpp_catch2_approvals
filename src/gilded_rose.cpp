#include "gilded_rose.hpp"

#include <items/item_converter.hpp>

GildedRose::GildedRose(std::vector<Item>& items)
{
    internalItems = convert(items);
}

void GildedRose::updateQuality()
{
    for (auto& item : internalItems) {
        item->update();
    }
}
