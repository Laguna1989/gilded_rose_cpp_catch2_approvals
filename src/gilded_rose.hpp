#include <string>
#include <vector>

class Item {
public:
    std::string name;
    int sellIn;
    int quality;

    Item(std::string const& name, int sellIn, int quality)
        : name(name)
        , sellIn(sellIn)
        , quality(quality)
    {
    }
};

class GildedRose {
public:
    explicit GildedRose(std::vector<Item>& items);

    void updateQuality();

private:
    std::vector<Item>& items;
};
