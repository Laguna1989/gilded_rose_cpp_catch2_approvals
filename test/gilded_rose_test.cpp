#include "gilded_rose.hpp"

#include <ApprovalTests.hpp>
#include <catch2/catch_test_macros.hpp>


TEST_CASE("Empty test case")
{
    REQUIRE(true);
}
