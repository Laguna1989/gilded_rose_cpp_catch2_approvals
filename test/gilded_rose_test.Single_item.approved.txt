name: Axe, sellIn: 9, quality: 14
